endpoint/crypto.go

```
@@ -0,0 +1,124 @@
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+/*
+This code is based on the article https://www.thepolyglotdeveloper.com/2018/02/encrypt-decrypt-data-golang-application-crypto-packages/
+written by Nic Raboy.
+Gzip compression/decompression has ben added as there as there are issues with txt records longer than
+255 characters and encryption + base64 encoding adds some characters to the resulting text record.
+*/
+
+package endpoint
+
+import (
+	"bytes"
+	"compress/gzip"
+	"crypto/aes"
+	"crypto/cipher"
+	"crypto/rand"
+	"encoding/base64"
+	"fmt"
+	"io"
+
+	log "github.com/sirupsen/logrus"
+)
+
+// Decompress data
+func DecompressData(data []byte) (resData []byte, err error) {
+	gz, err := gzip.NewReader(bytes.NewBuffer(data))
+	if err != nil {
+		return data, err
+	}
+	defer gz.Close()
+	var b bytes.Buffer
+	if _, err = b.ReadFrom(gz); err != nil {
+		return data, err
+	}
+	return b.Bytes(), nil
+}
+
+// Compress data
+func CompressData(data []byte) (compressedData []byte, err error) {
+	var b bytes.Buffer
+	gz, err := gzip.NewWriterLevel(&b, gzip.BestCompression)
+	if err != nil {
+		return data, nil
+	}
+	// defer gz.Close()
+	if _, err = gz.Write(data); err != nil {
+		return data, nil
+	}
+
+	if err = gz.Flush(); err != nil {
+		return data, nil
+	}
+
+	if err = gz.Close(); err != nil {
+		return data, nil
+	}
+
+	return b.Bytes(), nil
+}
+
+// Encrypt data using the supplied AES key
+func EncryptText(text string, aesKey []byte) (string, error) {
+	block, _ := aes.NewCipher(aesKey)
+	gcm, err := cipher.NewGCM(block)
+	if err != nil {
+		return text, err
+	}
+	nonce := make([]byte, gcm.NonceSize())
+	if _, err = io.ReadFull(rand.Reader, nonce); err != nil {
+		return text, err
+	}
+	data := []byte(text)
+	data, err = CompressData(data)
+	if err != nil {
+		log.Debugf("Failed to compress data based on the text %#v. Got error %#v.", text, err)
+	}
```

Are we good hiding the errors?